### ISSUE TITLE
Testing Firefox

### ISSUE THREAD

**Issue by CyrosX** [heeplr/document-dl#2]
- Mac OS 11.5.2
- Firefox 91.0.1 + Selenium

I'm running this command:

```
 document-dl -b firefox -u NUMBER -p 'PASSWORD' --action download --jq 'contains({id: 0})' o2
```
It starts a pure Firefox without anything installed (it's a new profile I think).
After a while it exits and I'm getting these errors:

```
Traceback (most recent call last):
  File "/usr/local/bin/document-dl", line 8, in <module>
    sys.exit(documentdl())
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/docdl/plugins/o2.py", line 133, in o2
    docdl.cli.run(ctx, O2)
  File "/usr/local/lib/python3.9/site-packages/docdl/cli.py", line 150, in run
    plugin = plugin_class(
  File "/usr/local/lib/python3.9/site-packages/docdl/__init__.py", line 147, in __init__
    self._init_webdriver(webdriver_opts, arguments['webdriver'])
  File "/usr/local/lib/python3.9/site-packages/docdl/__init__.py", line 272, in _init_webdriver
    self.webdriver = webdriver.Firefox(
  File "/usr/local/lib/python3.9/site-packages/selenium/webdriver/firefox/webdriver.py", line 190, in __init__
    executor = ExtensionConnection("127.0.0.1", self.profile,
  File "/usr/local/lib/python3.9/site-packages/selenium/webdriver/firefox/extension_connection.py", line 52, in __init__
    self.binary.launch_browser(self.profile, timeout=timeout)
  File "/usr/local/lib/python3.9/site-packages/selenium/webdriver/firefox/firefox_binary.py", line 73, in launch_browser
    self._wait_until_connectable(timeout=timeout)
  File "/usr/local/lib/python3.9/site-packages/selenium/webdriver/firefox/firefox_binary.py", line 109, in _wait_until_connectable
    raise WebDriverException(
selenium.common.exceptions.WebDriverException: Message: Can't load the profile. Possible firefox version mismatch. 
You must use GeckoDriver instead for Firefox 48+. Profile Dir: /var/folders/td/x4r_b40s2r5bvlrlrby13ptw0000gn/T/tmp799ugx0k 
If you specified a log_file in the FirefoxBinary constructor, check it for details.
``

**Comment by heeplr** [comment]
Could you try to remove [this line](https://github.com/heeplr/document-dl/blob/master/docdl/__init__.py#L273) and check if the error still persists?

I too couldn't get firefox/geckodriver to run with selenium after some updates and switched to chrome. I'm almost sure it's seleniums fault and will be fixed in future versions. If not, I'd happily merge a PR that fixes or works around it.
For now I'd recommend using chrome if you don't want to deep dive into debugging.

This simple testcase also fails for me:

```python
from selenium import webdriver
driver = webdriver.Firefox()
```
with:
```
selenium.common.exceptions.SessionNotCreatedException: Message: Unable to find a matching set of capabilities
```

if a capabilities argument is passed to the constructor, it results in the same error that you experience:

```python
driver = webdriver.Firefox(capabilities={})
```
yields:
```
selenium.common.exceptions.WebDriverException: Message: Can't load the profile. Possible firefox version mismatch. You must use GeckoDriver instead for Firefox 48+. Profile Dir: /tmp/tmpjzyzk_qj If you specified a log_file in the FirefoxBinary constructor, check it for details.
```

EDIT: added testcase

**Comment by CyrosX** [comment]
did not work either :-S. 
Is the GeckoDriver automatically installed?
I've tried it before & after installing it through brew.sh (brew install GeckoDriver)
but the result is the same.
Maybe it needs to be specified where the executable is?

**Comment by heeplr** [comment]
The cause is [this bug](https://github.com/mozilla/geckodriver/issues/1756) and I implemented a ~very bad~ temporary workaround in  c60e812a99. ~(This breaks on platforms where the firefox binary isn't in /usr/lib64/firefox)~.
This introduced/exposed quite a few new issues and I tried to fix them all.
Closing this. Please reopen if there are still issues.

EDIT: not that bad anymore
